Mathlib/Topology/ContinuousFunction/FunctionalCalculus.lean

```diff
@@ -0,0 +1,676 @@
+properties that it is a continuous star algebra homomorphism mapping the (restriction of) the
+identity to `a`. This is the necessary tool used to establish `cfcSpec_comp` and the more common
+variant `cfc_comp`.
+
+This class will have instances in each of the common cases `ℂ`, `ℝ` and `ℝ≥0` as a consequence of
+the Stone-Weierstrass theorem. -/
+class UniqueContinuousFunctionalCalculus (R A : Type*) [CommSemiring R] [StarRing R]
+    [MetricSpace R] [TopologicalSemiring R] [ContinuousStar R] [Ring A] [StarRing A]
+    [TopologicalSpace A] [Algebra R A] : Prop where
+  eq_of_continuous_of_map_id (a : A) (φ ψ : C(spectrum R a, R) →⋆ₐ[R] A)
+    (hφ : Continuous φ) (hψ : Continuous ψ)
+    (h : φ (.restrict (spectrum R a) <| .id R) = ψ (.restrict (spectrum R a) <| .id R)) :
+    φ = ψ
+
+variable {R A : Type*} {p : A → Prop} [CommSemiring R] [StarRing R] [MetricSpace R]
+variable [TopologicalSemiring R] [ContinuousStar R] [TopologicalSpace A] [Ring A] [StarRing A]
+variable [Algebra R A] [ContinuousFunctionalCalculus R p]
+
+lemma StarAlgHom.ext_continuousMap [UniqueContinuousFunctionalCalculus R A]
+    (a : A) (φ ψ : C(spectrum R a, R) →⋆ₐ[R] A) (hφ : Continuous φ) (hψ : Continuous ψ)
+    (h : φ (.restrict (spectrum R a) <| .id R) = ψ (.restrict (spectrum R a) <| .id R)) :
+    φ = ψ :=
+  UniqueContinuousFunctionalCalculus.eq_of_continuous_of_map_id a φ ψ hφ hψ h
+
+section cfcSpec
+
+variable {a : A} (ha : p a)
+
+-- Note: since `spectrum R a` is closed, we may always extend `f : C(spectrum R a, R)` to a function
+-- of type `C(R, R)` by the Tietze extension theorem (assuming `R` is either `ℝ`, `ℂ` or `ℝ≥0`).
+
+/-- The star algebra homomorphism underlying a instance of the continuous functional calculus;
+a version for continuous functions on the spectrum.
+
+In this case, the user must supply the fact that `a` satisfies the predicate `p`, for otherwise it
+may be the case that no star algebra homomorphism exists. For instance if `R := ℝ` and `a` is an
+element whose spectrum (in `ℂ`) is disjoint from `ℝ`, then `spectrum ℝ a = ∅` and so there can be
+no star algebra homomorphism between these spaces.
+
+While `ContinuousFunctionalCalculus` is stated in terms of these homomorphisms, in practice the
+user should instead prefer `cfc` over `cfcSpec`.
+-/
+def cfcSpec : C(spectrum R a, R) →⋆ₐ[R] A :=
+  ContinuousFunctionalCalculus.toStarAlgHom ha
+
+lemma cfcSpec_closedEmbedding :
+    ClosedEmbedding <| (cfcSpec ha : C(spectrum R a, R) →⋆ₐ[R] A) :=
+  ContinuousFunctionalCalculus.hom_closedEmbedding ha
+
+lemma cfcSpec_map_id :
+    cfcSpec ha ((ContinuousMap.id R).restrict <| spectrum R a) = a :=
+  ContinuousFunctionalCalculus.hom_id ha
+
+/-- The **spectral mapping theorem** for the continuous functional calculus. -/
+lemma cfcSpec_map_spectrum (f : C(spectrum R a, R)) :
+    spectrum R (cfcSpec ha f) = Set.range f :=
+  ContinuousFunctionalCalculus.hom_map_spectrum ha f
+
+lemma cfcSpec_predicate (f : C(spectrum R a, R)) :
```

Comment:
Sounds stupid, but this is actually the convention:
```suggestion
lemma cfcSpec_spec (f : C(spectrum R a, R)) :
```
cf `Classical.some_spec`, `Nat.find_spec`...

Reply:
I don't think that applies here. That naming convention applies when you apply some variant of `Classical.choice` and you want the predicate it satisfies. That's not what's going on here. In this case `predicate` refers to the `p` which is an outParam in `ContinuousFunctionalCalculus`.

Reply:
I disagree. See eg `ringChar.spec`, `maxChain_spec`, `Group.rank_spec` for more examples where `_spec` is used to indicate that some constructed object respects some property.

Reply:
I agree that this is not specific to things obtained by choice, but I do think it doesn't apply here: `spec` stands for "specification", all of these lemmas are the defining property of some object, which isn't the case here.